[CS2113-W12-1] HealthMate

DarkDragoon2002 opened the pull request:
Helping busy and CLI-affine people to build healthy eating habits to improve their mental and physical well-being.

wenxin-c commented on src/main/java/seedu/healthmate/HealthGoal.java:
```diff
@@ -0,0 +1,102 @@
+     * @param healthGoalInput the input health goal (e.g., WEIGHT_LOSS).
+     */
+    public void saveHealthGoal(String healthGoalInput) {
+        assert healthGoalInput != null : "Health goal input cannot be null";
```
Good use of assertion!!

wenxin-c commented on src/main/java/seedu/healthmate/HealthGoal.java:
```diff
@@ -0,0 +1,102 @@
+        if (isMale) {
+            rawCaloriesTarget = 88.362 + (13.397 * weight) + (4.799 * height) - (5.677 * age);
+        } else {
+            rawCaloriesTarget = 447.593 + (9.247 * weight) + (3.098 * height) - (4.330 * age);
```
You could consider turning the magic numbers into constant as well  to give it a meaning.

wenxin-c commented on src/main/java/seedu/healthmate/DateTimeUtils.java:
```diff
@@ -0,0 +1,52 @@
+     * @return the current LocalDate
+     */
+    public static LocalDate currentDate() {
+        return LocalDate.now();
```
You could try to name methods as verbs instead of noun, e.g. ```getCurrentData()```

wenxin-c commented on src/main/java/seedu/healthmate/HistoryTracker.java:
```diff
@@ -0,0 +1,196 @@
+            System.out.println("Error parsing a number." + e.getMessage());
+        } catch (NoSuchElementException e) {
+            // silent catch if existing user file contains no content
+        }
```
Try not to use empty catch block, you might want to implement some error catching/logging in the catch block.

wenxin-c commented on src/main/java/seedu/healthmate/HistoryTracker.java:
```diff
@@ -0,0 +1,196 @@
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
```
This class is getting very long. One improvement you can consider is to extract such helper methods e.g. saving data to a file etc into a ```utils``` class and invoke relevant methods if needed.

wenxin-c commented on src/main/java/seedu/healthmate/UI.java:
```diff
@@ -0,0 +1,207 @@
+            printReply("No meal entries added yet", "");
+        }
+        printSeparator();
+    }
```
These two methods are pretty similar, you could avoid repeating yourself by combining them into one and pass in parameters to differentiate scenarios.

wenxin-c commented on src/main/java/seedu/healthmate/command/commands/AddMealEntryCommand.java:
```diff
@@ -0,0 +1,16 @@
+    private static final String FORMAT =
+            "add mealEntry {meal name from menu} | or | add mealEntry {meal name} /c{number of " + "calories}";
+    private static final String DESCRIPTION = "Adds a meal entry to the meal log either from a pre-existing meal in " +
+            "the" +
```
You could consider splitting the line at other places instead of having a single word on one line.

wenxin-c commented on src/main/java/seedu/healthmate/MealList.java:
```diff
@@ -0,0 +1,96 @@
+
+    public int size() {
+        return this.mealList.size();
+    }
```
Try to name this method as verb as well like other methods, e.g. ```getSize()```

wenxin-c commented on src/main/java/seedu/healthmate/MealSaver.java:
```diff
@@ -0,0 +1,50 @@
+
+    public Meal extractMealFromUserInput(String userInput) {
+        try {
+            String command = "save meal";
```
You might consider extracting some string into named constant to give it some meaning.

wenxin-c commented on src/main/java/seedu/healthmate/ChatParser.java:
```diff
@@ -0,0 +1,172 @@
+        while (!userInput.equals("bye")) {
+            logger.log(Level.INFO, "Getting next user input line");
+            userInput = scanner.nextLine().strip();
+            switch (userInput) {
+            case "bye":
+                logger.log(Level.INFO, "User closes application");
+                UI.printFarewell();
+                break;
+            default:
+                try {
+                    this.multiCommandParsing(userInput, user);
+                    logger.log(Level.INFO, "User input contains more than 1 token");
+                } catch (ArrayIndexOutOfBoundsException a) {
+                    logger.log(Level.WARNING, "Invalid command", a);
+                    UI.printReply("Invalid command", "Retry: ");
+                }
```
If there are no more potential cases to be added, you could consider just use a simple ```if...else...```. ```switch``` will be more suitable if there are more cases.

wenxin-c commented on src/main/java/seedu/healthmate/ChatParser.java:
```diff
@@ -0,0 +1,172 @@
+package seedu.healthmate;
+
+import seedu.healthmate.command.Command;
+import seedu.healthmate.command.commands.LogMealsCommand;
+import seedu.healthmate.command.commands.SaveMealCommand;
+import seedu.healthmate.command.commands.ListCommandsCommand;
+import seedu.healthmate.command.commands.AddMealEntryCommand;
+import seedu.healthmate.command.commands.DeleteMealCommand;
+import seedu.healthmate.command.commands.DeleteMealEntryCommand;
+import seedu.healthmate.command.commands.MealMenuCommand;
+import seedu.healthmate.command.commands.UpdateUserDataCommand;
+import seedu.healthmate.command.commands.TodayCalorieProgressCommand;
+import seedu.healthmate.command.commands.HistoricCalorieProgressCommand;
+import seedu.healthmate.command.CommandMap;
```
Good extraction of commands!!

wenxin-c reviewed (commented):
Overall good work from the team. Some minor suggestions for you to consider. Keep it up!!

cnivedit commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+To create or load a user profile the `User` class provides the method `checkForUserData` which loads 
+saved user information if available from an existing file usingn the `HistoryTracker` or prompts the user
+to input new information for creating a new profile as shown in the sequence diagram below.
+![User SD](images/userSequenceDiagram.jpg)
```
<img width="530" alt="image" src="https://github.com/user-attachments/assets/92753e13-d39e-4beb-aa87-5f3b553c0de0">
Does the scanner instance exist prior to this point? If yes, there shouldnt be an activation bar for the constructor and in case it was newly created, perhaps there should be a call to the constructor?

cnivedit replied:
![image](https://github.com/user-attachments/assets/3c0fbdf7-32e8-43f0-98ef-b242a7b5187e)
Perhaps consider omitting info about file creation as it may not be relevant to the purpose of the sequence diagram

cnivedit commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
+printing output to the user.
+
+![High Level CD](images/highLevelClassDiagram.jpg)
```
![image](https://github.com/user-attachments/assets/c92b2b27-04bc-4568-900f-17e7fa37d51a)

Should the class names be preceded with a colon?

kaboomzxc replied:
Good separation of concerns between classes !

cnivedit commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+The Meal class encapsulates the concept of a meal. As the purpose of this application 
+is to track calorie consumption, this consists of a mandatory calorie entry. The meal's name attribute, 
+is however an Optional<String> allowing a case, where no meaningful label can be attached to a certain consumption.
+![User SD](images/addMeal.png)
```
![image](https://github.com/user-attachments/assets/89f6cdf8-f8f7-43c3-a74b-27381089b995)
Image fails to load

kennethszj replied:
![image](https://github.com/user-attachments/assets/7b0d015d-9fee-4fb2-a5f5-d3f5e0c71f35)
the sequence for extractAndAppendMeal() is clear in showing that a meal object is created and appended. 
Would it be clearer to describe what information is extracted? like what does this method extract (eg, name, calorie...)

cnivedit commented on docs/images/addMeal.png:
Should the sequence diagram omit program execution prior to when add meal command is detected?

cnivedit reviewed (commented):
Good attempt! Consider refining and adding more diagrams

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
+printing output to the user.
```
although I understand that you mentioned Ui class is omitted for clarity, however my modest opinion is that Ui is crucial component to be documented. would a separate UI class diagram be useful?

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -27,12 +227,52 @@
+   - Restart the application and check if previously added meals and logged entries are still present.
+
+8. Error Handling:
+   - Try entering invalid commands or data to ensure the application handles errors gracefully and provides helpful error messages.
```
would it be favourable to show some examples?

kaboomzxc commented on src/main/java/seedu/healthmate/HealthGoal.java:
```diff
@@ -0,0 +1,102 @@
+    public double getTargetCalories(double height, double weight, boolean isMale, int age) {
+        assert height > 0 : "Height must be positive";
+        assert weight > 0 : "Weight must be positive";
+        assert age > 0 : "Age must be positive";
```
good to guard against negative height/weight/age ! 
I am wondering, would it be useful to guard against unrealistic data such as 500 yrs of Age ?

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+printing output to the user.
+
+![High Level CD](images/highLevelClassDiagram.jpg)
+
+#### HealthMate
```
Clear explanation of classes

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+
+HistoryTracker allows for the persistance of user inputted data between sessions by storing it in a local csv file.
+
+## Features
```
Could some Sequence Diagrams of the processing of Commands be shown?

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+![User SD](images/userSequenceDiagram.jpg)
+Reference diagrams used
+![User SD](images/askForUserDataSD.png)
+![Read User SD](images/readUserDataFromFileSD.png)
```
Good decision to show two sub-diagrams for clarity; creating a new profile for new user, and loading saved file for existing user

kaboomzxc commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+---
+### Creating a User Profile
+To create or load a user profile the `User` class provides the method `checkForUserData` which loads 
+saved user information if available from an existing file usingn the `HistoryTracker` or prompts the user
```
found a minor typo hehe , "using" is the correct spelling instead of "usingn"

kaboomzxc reviewed (commented):
Overall DG is okay good 👍🏻

kennethszj commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+This specifically includes adherance to the "Tell Don't Ask" principle which was enforced by
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
```
would it be helpful to include a minimal/simple representation of the UI class in the high level class diagram to indicate dependency across all classes. this might improve clarity around UI interactions

kennethszj commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+To create or load a user profile the `User` class provides the method `checkForUserData` which loads 
+saved user information if available from an existing file usingn the `HistoryTracker` or prompts the user
+to input new information for creating a new profile as shown in the sequence diagram below.
+![User SD](images/userSequenceDiagram.jpg)
```
i agree with cnivedit about the Scanner instance's activation bar. additionally, if scanner is indeed reused across multiple interactions, would it make sense to show its initialisation outside the main sequence flow?

kennethszj commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,222 @@
+to input new information for creating a new profile as shown in the sequence diagram below.
+![User SD](images/userSequenceDiagram.jpg)
+Reference diagrams used
+![User SD](images/askForUserDataSD.png)
```
![image](https://github.com/user-attachments/assets/c23d55b8-3f88-408c-a664-ac73dee3e55f)

in the sequence, "printString(...)" is used multiple times. would it be more readable to differentiate each print string with specific text or parameters? (eg, printstring(enter calorie:)). this could help viewers understand what actual prompt is displayed at each step.

jadenlimjc commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,288 @@
+This specifically includes adherance to the "Tell Don't Ask" principle which was enforced by
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
```
likewise to previous reviewers, my belief is that some documentation of the UI component would be useful in order to provide clarity into possible outputs a user should receive. perhaps you could separate the UI diagram form the overarching diagram provided to retain the clarity your diagram currently has while still providing viewers with the information regarding the UI class.

jadenlimjc commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,288 @@
+- A `MealList` object called `mealOptions` 
+  - Contains meals that are presaved by the user for quick selection to track commonly consumed meals
+  - the command `meal menu` is used to display the current mealOptions. The implementation of this command is shown in the UML diagram below.
+![Meal Menu SD](images/mealMenuSD.png)
```
perhaps the diagram could include a user so that the run() function does not appear out of nowhere, providing additional clarity.

jadenlimjc commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,288 @@
+To create or load a user profile the `User` class provides the method `checkForUserData` which loads 
+saved user information if available from an existing file usingn the `HistoryTracker` or prompts the user
+to input new information for creating a new profile as shown in the sequence diagram below.
+![User SD](images/userSequenceDiagram.jpg)
```
personally, this image used is quite blurry and  the details are hard to see regardless of zoom. perhaps the team could consider making the diagram slightly easier on the eyes to improve user experience. another possible solution could be to break the diagram into smaller independent sections. as cnivedit mentioned, the file creation may not be the mos relevant to the creation of a user profile so can consider depicting file creation in a separate diagram.

jadenlimjc reviewed (commented):
Overall a good DG, perhaps the Command Handling Section could do with some sequence diagrams or examples of inputs to aid clarity

ryan-txn committed:
Remove unecessary import

ryan-txn committed:
Update user dummy data

ryryry-3302 committed:
Merge pull request #109 from ryan-txn/Implement_Single_User_TimeStamp

Add UserEntryList class. Change txt save file to csv.

ryryry-3302 committed:
Create recommender package

ryryry-3302 committed:
Merge branch 'AY2425S1-CS2113-W12-1:master' into 66-add-meal-option-recommendations

ryryry-3302 committed:
create recipe map

ryryry-3302 committed:
create meal recommendations

ryryry-3302 committed:
Test: create test cases

ryryry-3302 committed:
Fix styles

ryryry-3302 committed:
Add: Update commands

ryryry-3302 committed:
Feat: Add Weight timeline display

ryryry-3302 committed:
Test: Weight Entry Display test

kennethSty committed:
Merge pull request #123 from ryryry-3302/66-add-meal-option-recommendations

66 add meal option recommendations

kennethSty committed:
Bug fix historicCalories Test

kennethSty committed:
update DG and UG

kennethSty committed:
Fix test

ryryry-3302 committed:
Merge pull request #126 from AY2425S1-CS2113-W12-1/updateDG

Update DG

ryryry-3302 committed:
Merge branch 'master' into 60-timeline-of-weight-history

Dri-water committed:
Fix timestamp bug

ryryry-3302 committed:
Style: fix styling issue

ryryry-3302 committed:
Merge branch '60-timeline-of-weight-history' of https://github.com/ryryry-3302/tp into 60-timeline-of-weight-history

ryryry-3302 committed:
Merge branch 'master' into fixTest

ryryry-3302 committed:
Merge pull request #124 from AY2425S1-CS2113-W12-1/fixTest

Bug fix historicCalories Test

ryryry-3302 committed:
Merge pull request #128 from Dri-water/fix-timestamp

Fix timestamp bug

ryryry-3302 committed:
Merge branch 'master' into 60-timeline-of-weight-history

Dri-water committed:
Add JavaDoc header comments

Dri-water committed:
Merge pull request #127 from ryryry-3302/60-timeline-of-weight-history

60 timeline of weight history

ryryry-3302 committed:
Merge pull request #129 from Dri-water/header-comments

Add JavaDoc header comments

DarkDragoon2002 committed:
Docs: Claiming Worked on Code Lines

-KennethSty helped to abstract my code on historicCalories Command
-Claiming only Core parts of the code and logic I worked on
-Did not claim most helper functions
-Also claimed the delete mealEntry and delete meal commands which I worked on

DarkDragoon2002 committed:
Task: Add Assertions to make ChatParser more Defensive

DarkDragoon2002 committed:
Task: Abstracted Chat Parser and shifted Code to respective Commands

DarkDragoon2002 committed:
Docs: JavaDoc Comments for all the Command Classes

DarkDragoon2002 committed:
Bug: Minor Bug Fix

DarkDragoon2002 committed:
Test: Added Tests for Delete Commands and Today Calories

DarkDragoon2002 committed:
Fix: CheckStyle fixes

DarkDragoon2002 committed:
Fix: Testing Bug Fix

kennethSty committed:
Add author tags to UserHistoryTracker

kennethSty committed:
Make code more readable

kennethSty committed:
Merge pull request #130 from DarkDragoon2002/Defensive-Coding

Defensive coding

kennethSty committed:
Add more logging and make use of user stub in chatparser tests

kennethSty committed:
Add test for extracting timestamp from user input

kennethSty committed:
Add fail assertion to portion test

kennethSty committed:
Fix checkstyle issue

kennethSty committed:
Fix import checkstyle violation

kennethSty committed:
Add author tags

kennethSty committed:
Merge branch 'master' into addAuthorTags

kennethSty committed:
Resolve more conflicts for divergent upstream and local branch

Dri-water committed:
Merge pull request #131 from AY2425S1-CS2113-W12-1/addAuthorTags

Add author tags to UserHistoryTracker

Dri-water committed:
Merge pull request #133 from AY2425S1-CS2113-W12-1/AddTags

Add author tags

Dri-water committed:
Merge branch 'master' into PolishCode

Dri-water committed:
Merge pull request #132 from AY2425S1-CS2113-W12-1/PolishCode

Add more logging and make use of user stub in chatparser tests

ryan-txn committed:
Add JavaDoc Comments

kennethSty committed:
Merge pull request #134 from ryan-txn/AddJavaDocComments

Add JavaDoc Comments

kennethSty committed:
Remove unnecessary double reference to history trackers

kennethSty committed:
Clean up ChatParser

kennethSty committed:
Add javadoc comments

kennethSty committed:
Remove duplication and SLAP UserHistoryTracker methods

kennethSty committed:
Update SD to create a new user profile

kennethSty committed:
Fix checkstyle

kennethSty committed:
Apply single responsibility principle to User class

kennethSty committed:
Do houskeeping on core classes

ryan-txn committed:
Add User History Tracker Test class

Add basic test for checking correct adding of a User. Add setup and cleanup of test class.

ryan-txn committed:
Merge pull request #135 from AY2425S1-CS2113-W12-1/updateSQD

Update Sequence Diagram, and clean code

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into AddUserDataJUnitTests

ryan-txn committed:
Add javadoc comments and author tags

kennethSty committed:
Improve Cohesion and add updated class diagram

kennethSty committed:
Update DG

kennethSty committed:
Delete unused import

ryan-txn committed:
Add Loading userEntryList JUnit test

kennethSty committed:
Update PPP

ryan-txn committed:
Merge pull request #136 from AY2425S1-CS2113-W12-1/updateHighLevelClassD

Update high level class diagram and improve cohesion

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into AddUserDataJUnitTests

# Conflicts:
#	src/main/java/seedu/healthmate/services/UserHistoryTracker.java

ryan-txn committed:
Fix Checkstyle

ryryry-3302 committed:
Merge pull request #137 from ryan-txn/AddUserDataJUnitTests

Add user data Junit tests

ryryry-3302 committed:
Delete docs/images/askForUserDataSD.png

ryryry-3302 committed:
Add files via upload

ryryry-3302 committed:
Update DeveloperGuide.md

ryryry-3302 committed:
Delete docs/images/readUserDataFromFileSD.png

ryryry-3302 committed:
Update DeveloperGuide.md

ryryry-3302 committed:
Test: make code more defensive

ryryry-3302 committed:
Bug: fix for save meal

ryryry-3302 committed:
style

ryryry-3302 committed:
Styling

ryan-txn committed:
Merge pull request #138 from AY2425S1-CS2113-W12-1/121-issue-in-askforuserdatasd

121 issue in askforuserdatasd

ryan-txn committed:
Merge pull request #140 from ryryry-3302/139-fix-and-misc-bugs

139 fix and misc bugs

ryan-txn committed:
Add Load User Entries SD

ryan-txn committed:
Add CurrentUserDataCommand

Update CommandMap. Update UG. Update DG.

ryan-txn committed:
Update CommandMap test case

ryryry-3302 committed:
Merge pull request #142 from ryan-txn/AddCurrentUserDataCommand

Add CurrentUserDataCommand

ryryry-3302 committed:
Merge pull request #141 from ryan-txn/AddLoadUserEntryListSD

Add Load User Entries SD

ryryry-3302 committed:
Release: plus fix

ryryry-3302 committed:
Release

ryryry-3302 committed:
Fix

kennethSty committed:
Merge pull request #144 from ryryry-3302/143-make-release

143 make release

ryryry-3302 committed:
Update UG

kennethSty committed:
Add new commands and fix errors

kennethSty committed:
Merge pull request #146 from ryryry-3302/145-update-userguide

Update UG

kennethSty committed:
Clean data files

ryan-txn committed:
Add CreateFilleIfNotExists SD

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into AddCurrentUserDataCommand

# Conflicts:
#	docs/DeveloperGuide.md

ryryry-3302 committed:
Merge pull request #148 from ryan-txn/AddCreateFileIfNotExistsSD

Add create file if not exists SD

ryryry-3302 committed:
Fix ug

kennethSty committed:
Replace special characters to be stable in windows

kennethSty committed:
Merge branch 'master' into AddNewCommandsUG

ryryry-3302 committed:
Merge pull request #147 from AY2425S1-CS2113-W12-1/AddNewCommandsUG

Add new commands and fix errors

Dri-water committed:
Fix mealMenu sequence diagram

ryryry-3302 committed:
Merge branch 'master' into 145-update-userguide

ryryry-3302 committed:
Merge pull request #150 from Dri-water/mealMenuSD-update

Fix mealMenu sequence diagram

Dri-water committed:
Merge pull request #149 from ryryry-3302/145-update-userguide

Fix ug

ryryry-3302 committed:
Fix: add identation

kennethSty committed:
Merge pull request #151 from ryryry-3302/145-update-userguide

Fix: add identation

ryryry-3302 committed:
add sequence diagram

ryryry-3302 committed:
Update dg

DarkDragoon2002 committed:
Bug: Gender Check Bug Fix in User Creation

-Non Male entry in user was automatically assigned to female
-Added a function to check

DarkDragoon2002 committed:
Bug: Delete Commands gave wrong Error when file was empty

DarkDragoon2002 committed:
Fix: Fixed Testcases for delete commands

DarkDragoon2002 committed:
Bug: Fixed add mealEntry with empty label

DarkDragoon2002 committed:
Bug + Doc: Fixed bug that allowed meal/mealEntry with "," to be saved and updated DG

DarkDragoon2002 committed:
Doc: Replaced And symbol with "and"

DarkDragoon2002 committed:
Update DeveloperGuide.md

DarkDragoon2002 committed:
Merge pull request #153 from ryryry-3302/152-update-dg

add sequence diagram

ryryry-3302 committed:
Merge pull request #154 from DarkDragoon2002/Update-SD-and-Bug-Fix

Bug fix  and Update DG

ryryry-3302 committed:
Fix issue 197

ryryry-3302 committed:
Fix typo in #182

kennethSty committed:
Add silent load for syncing data across multiple runs

ryryry-3302 committed:
Fix: #181

kennethSty committed:
Fix todayCalorieProgress test by ensuring implicit empty meal assumption is met

ryryry-3302 committed:
Fix: #189 non updating user data

ryryry-3302 committed:
Merge pull request #207 from AY2425S1-CS2113-W12-1/FixMultipleInstanceBug

Add silent load for syncing data across multiple runs

ryryry-3302 committed:
Merge branch 'master' into 197-fixing-easy-cosmetic-issues

ryryry-3302 committed:
Merge pull request #208 from ryryry-3302/197-fixing-easy-cosmetic-issues

197 fixing easy cosmetic issues

ryryry-3302 committed:
Fix: #172

ryan-txn committed:
Add max exceptions for height, weight and age

kennethSty committed:
Fix data corruption bug

kennethSty committed:
Merge pull request #209 from ryan-txn/AddRealisticUserParams

Add max exceptions for height, weight and age

ryan-txn committed:
Merge pull request #210 from AY2425S1-CS2113-W12-1/FixDataManipulationBug

Fix data corruption bug

kennethSty committed:
Fix 0 portion bug

kennethSty committed:
Save log

kennethSty committed:
Merge master into BugFix branch

kennethSty committed:
Fix checkstyle issue

kennethSty committed:
Delete unused import

Dri-water committed:
Merge pull request #211 from AY2425S1-CS2113-W12-1/FixZeroPortionsBug

Fix zero portions bug

ryan-txn committed:
Add lower limit to HistoricCalorieProgressCommand number of days

kennethSty committed:
Fix Calories Bug

kennethSty committed:
Improve duplicate meal option feedback and make duplication detection not case sensitive

kennethSty committed:
Update UG

kennethSty committed:
Merge pull request #212 from ryan-txn/AddHistoricCalorieProgressCommandParamLimits

Add lower limit to HistoricCalorieProgressCommand number of days

ryryry-3302 committed:
Merge pull request #213 from AY2425S1-CS2113-W12-1/CapitalLetterAndCaloriesBug

Update UG and improve meal option update UX

kennethSty committed:
Update UG with correct log meals output

Dri-water committed:
Fix HistoryTracker error messages

ryryry-3302 committed:
Merge pull request #214 from AY2425S1-CS2113-W12-1/TimeFormatBug

Update UG with correct log meals output

ryryry-3302 committed:
Fix: #179

ryryry-3302 committed:
Bug: fix small bug related to #179

Dri-water committed:
Add early return if no csv file in HistoryTracker

Dri-water committed:
Merge pull request #215 from Dri-water/fix-HistoryTracker-error

Fix HistoryTracker error messages

kennethSty committed:
Decentralize asking for user data to enable quicker recovery from false input

kennethSty committed:
Fix userHistoryTracker test

ryryry-3302 committed:
Fix: #180

kennethSty committed:
Merge branch 'master' into ImproveAskForUserData

ryryry-3302 committed:
Fix merge conflict

ryryry-3302 committed:
Merge branch 'master' into 172-pe-dtester-a-explanation-too-brief-for-weight-timeline

kennethSty committed:
Remove unused imports

kennethSty committed:
Test run

ryryry-3302 committed:
Fix merge conflict

ryryry-3302 committed:
Merge branch 'master' into 172-pe-dtester-a-explanation-too-brief-for-weight-timeline

# Conflicts:
#	data/user_data.csv
#	logs/seedu.healthmate.services.ChatParser.log

Dri-water committed:
Merge pull request #216 from AY2425S1-CS2113-W12-1/ImproveAskForUserData

Improve ask for user data

ryryry-3302 committed:
Merge branch 'master' into 172-pe-dtester-a-explanation-too-brief-for-weight-timeline

ryryry-3302 committed:
Fix

ryryry-3302 committed:
Merge pull request #217 from ryryry-3302/172-pe-dtester-a-explanation-too-brief-for-weight-timeline

172 pe dtester a explanation too brief for weight timeline

kennethSty committed:
Add fallsback if progressbar characters unknown

kennethSty committed:
Make progressbar universally stable by choosing standard characters

kennethSty committed:
Merge branch 'master' into ProgressBarFallback

kennethSty committed:
Fix tests

ryryry-3302 committed:
Merge pull request #218 from AY2425S1-CS2113-W12-1/ProgressBarFallback

stabilize progress bar

kennethSty committed:
Add SQD, make UI system specific

kennethSty committed:
Delete unused imports

kennethSty committed:
Include user file for IO test

kennethSty committed:
Fix IO redirection test

ryryry-3302 committed:
Merge pull request #219 from AY2425S1-CS2113-W12-1/UpdateSDDG_ProgUG

Add SQD, make UI system specific

ryan-txn committed:
Add User Guide Content Page

ryan-txn committed:
Correct example used

ryryry-3302 committed:
Fix: Increase resolution of graph and add note in ug

ryryry-3302 committed:
Merge pull request #220 from ryan-txn/AddUserGuideContentsPage

Add User Guide Content Page

ryryry-3302 committed:
Fix: #196

ryryry-3302 committed:
Fix: #193

ryryry-3302 committed:
Fix: #168

ryryry-3302 committed:
Style

kennethSty committed:
Make name Optional for MealEntries for fast typists

ryryry-3302 committed:
Merge pull request #223 from AY2425S1-CS2113-W12-1/MakeNameOptionalAgain

Make name Optional for MealEntries for fast typists

kennethSty committed:
Merge pull request #222 from ryryry-3302/185-pe-dtester-a-the-features-are-repeated-in-quick-start-and-features

Fix #196, #193, #168

kennethSty committed:
Merge pull request #221 from ryryry-3302/183-pe-dtester-c-weight-timeline-fails-to-show-formal-small-changes-clearly-when-new-weight-change-is-large

Fix: Increase resolution of graph and add note in ug

ryryry-3302 committed:
update PPP

kennethSty committed:
Merge pull request #226 from ryryry-3302/224-ryan-update-ppp

update PPP

kennethSty committed:
Improve user feedback when user profile corrupted

kennethSty committed:
Update IO test

ryan-txn committed:
Strengthen User Data Parsing

Add more checks for parsing each line in user data file. Add clearing of incorrect data.

ryan-txn committed:
Merge pull request #227 from AY2425S1-CS2113-W12-1/RecoverUserAfterModifyingData

Improve user feedback when user profile corrupted

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into StrengthenUserDataParsingCode

# Conflicts:
#	src/main/java/seedu/healthmate/services/UserHistoryTracker.java

ryan-txn committed:
Touch up message formats

ryan-txn committed:
Remove unneeded imports, Reorder static variables

ryryry-3302 committed:
Merge pull request #228 from ryan-txn/StrengthenUserDataParsingCode

Strengthen user data parsing code

ryan-txn committed:
Add PPP

ryan-txn committed:
Add Clear User Data Command

Dri-water committed:
Fix mealMenu SD

ryryry-3302 committed:
Merge pull request #229 from ryan-txn/AddRyanTxnPPP

Add PPP

ryryry-3302 committed:
Merge pull request #231 from Dri-water/fix-mealMenuSD

Fix mealMenu SD

DarkDragoon2002 committed:
Bug: Added Max Desc Length Check to meals and mealEntries

ryryry-3302 committed:
Fix: #192

ryryry-3302 committed:
Fix:

DarkDragoon2002 committed:
Bug: Fixed target calories to be integer + more user friendly input for healthgoal

ryryry-3302 committed:
Style: fix checkstyle

DarkDragoon2002 committed:
Bug: Disallowed Future Dates from being added in mealEntry

DarkDragoon2002 committed:
Bug + Docs: Fixed ordering list commands and updated User Guide

ryan-txn committed:
Remove unused import

ryan-txn committed:
Merge remote-tracking branch 'upstream/master' into AddClearUserDataCommand

DarkDragoon2002 committed:
Fix: CheckStyle Fixes

ryan-txn committed:
Merge pull request #232 from ryryry-3302/make-commands-user-friendly-and-non-case-sensitive

Make commands user friendly and non case sensitive

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into AddClearUserDataCommand

# Conflicts:
#	src/main/java/seedu/healthmate/services/ChatParser.java

ryan-txn committed:
Merge pull request #230 from ryan-txn/AddClearUserDataCommand

Add Clear User Data Command

DarkDragoon2002 committed:
Merge branch 'master' into Bug-Fixes-+-Update-Docs

Dri-water committed:
Update Jonas PPP

DarkDragoon2002 committed:
Fix: Text-ui-test fixed

DarkDragoon2002 committed:
Merge branch 'Bug-Fixes-+-Update-Docs' of https://github.com/DarkDragoon2002/tp into Bug-Fixes-+-Update-Docs

Dri-water committed:
Fix PPP formatting

ryryry-3302 committed:
Merge pull request #234 from Dri-water/update-Jonas-PPP

Update Jonas PPP

DarkDragoon2002 committed:
Fix: Fixed user_data.csv

ryryry-3302 committed:
Merge pull request #233 from DarkDragoon2002/Bug-Fixes-+-Update-Docs

Bug fixes + Update docs

ryryry-3302 committed:
Fix: add documentation to add meal entry

kennethSty committed:
Update PPP

DarkDragoon2002 committed:
Docs: Updated Docs

kennethSty committed:
Merge pull request #236 from ryryry-3302/235-add-mealentry-command-missing-t-in-list-commands-documentation

Fix: add documentation to add meal entry

kennethSty committed:
Merge branch 'master' into UpdatePPPKenneth

ryryry-3302 committed:
Merge pull request #238 from AY2425S1-CS2113-W12-1/UpdatePPPKenneth

Update PPP

ryryry-3302 committed:
Merge pull request #237 from DarkDragoon2002/Update-Docs

PPP + DG changes

kennethSty committed:
change team name to lowercase

DarkDragoon2002 committed:
Fix: Renamed PPP according to submission guidelines

DarkDragoon2002 committed:
Fix: PPP Formatting

kennethSty committed:
Shorten PPP

kennethSty committed:
Merge pull request #239 from DarkDragoon2002/Update-Docs

Fix: Renamed PPP according to submission guidelines

ryryry-3302 committed:
Merge pull request #240 from AY2425S1-CS2113-W12-1/makeNameLower

change team name to lowercase

kennethSty committed:
Fix final errors in PPP

kennethSty committed:
Add short description to HP

kennethSty committed:
Merge pull request #241 from AY2425S1-CS2113-W12-1/makeNameLower

Include HP intro text and PPP updates

kennethSty committed:
Fix formatting UG

kennethSty committed:
Correct errors, added details

ryryry-3302 committed:
Merge pull request #242 from AY2425S1-CS2113-W12-1/UpdateDGKenneth

Fix formatting UG

ryryry-3302 committed:
Fix

ryryry-3302 committed:
Fix style

ryryry-3302 committed:
Update test

ryryry-3302 committed:
Merge pull request #243 from ryryry-3302/235-add-mealentry-command-missing-t-in-list-commands-documentation

Fix
